patches/net/minecraft/world/item/ItemStack.java.patch

```diff
@@ -1,30 +1,33 @@
 --- a/net/minecraft/world/item/ItemStack.java
 +++ b/net/minecraft/world/item/ItemStack.java
-@@ -81,7 +_,8 @@
+@@ -81,12 +_,13 @@
  import net.minecraft.world.level.block.state.pattern.BlockInWorld;
  import org.slf4j.Logger;
  
 -public final class ItemStack {
 +public final class ItemStack extends net.neoforged.neoforge.attachment.AttachmentHolder implements net.neoforged.neoforge.common.extensions.IItemStackExtension {
-+    // TODO: add attachment support here
      public static final Codec<ItemStack> CODEC = RecordCodecBuilder.create(
          p_311716_ -> p_311716_.group(
                      BuiltInRegistries.ITEM.holderByNameCodec().fieldOf("id").forGetter(ItemStack::getItemHolder),
-@@ -101,10 +_,12 @@
+                     Codec.INT.fieldOf("Count").forGetter(ItemStack::getCount),
+-                    CompoundTag.CODEC.optionalFieldOf("tag").forGetter(p_281115_ -> Optional.ofNullable(p_281115_.getTag()))
++                    CompoundTag.CODEC.optionalFieldOf("tag").forGetter(p_281115_ -> Optional.ofNullable(p_281115_.getTag())),
++                    ExtraCodecs.strictOptionalField(net.neoforged.neoforge.common.crafting.CraftingHelper.TAG_CODEC, ATTACHMENTS_NBT_KEY).forGetter(s -> Optional.ofNullable(s.serializeAttachments()))
```

Comment:
Do we want this to be `CraftingHelper.TAG_CODEC` or similar to the `tag` field about use `CompoundTag.CODEC`? The main reason I bring this up is while I agree having a broader supported set of inputs is nicer, because of the order of the `withAlternative` for `CraftingHelper.TAG_CODEC` it does mean that we will serialize this using `TagParger.AS_CODEC`, which is a different format than how the `tag` will be serialized as that only uses `CompoundTag.CODEC`.

Reply:
Done, and added a patch for ADVANCEMENT_ICON as well.